src/Tribe/Attendee_Registration/Service_Provider.php

```diff
@@ -8,36 +8,112 @@ class Tribe__Tickets__Attendee_Registration__Service_Provider extends tad_DI52_S
 	 * @since 4.9
 	 */
 	public function register() {
-		// Only run the code if ET+ is active.
+		$this->container->singleton( 'tickets.attendee_registration', Tribe__Tickets__Attendee_Registration__Main::class );
+		$this->container->singleton( 'tickets.attendee_registration.template', Tribe__Tickets__Attendee_Registration__Template::class );
+		$this->container->singleton( 'tickets.attendee_registration.view', Tribe__Tickets__Attendee_Registration__View::class );
+		$this->container->singleton( 'tickets.attendee_registration.rewrite', Tribe__Tickets__Attendee_Registration__Rewrite::class );
+		$this->container->singleton( 'tickets.attendee_registration.meta', Tribe__Tickets__Attendee_Registration__Meta::class );
+		$this->container->singleton( 'tickets.attendee_registration.shortcode', Tribe__Tickets__Attendee_Registration__Shortcode::class );
+		$this->container->singleton( 'tickets.attendee_registration.modal', Tribe__Tickets__Attendee_Registration__Modal::class );
+
+		// Priority 45 to be before priority 50 used on the same action in hooks().
+		add_action( 'tribe_plugins_loaded', [ $this, 'hooks' ], 45 );
+	}
+
+	/**
+	 * Add actions and filters for the Attendee Info classes.
+	 *
+	 * @since 4.9
+	 * @since TBD Bail if ETP is not active.
+	 */
+	public function hooks() {
+		// Only run the code if Event Tickets Plus is active.
 		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
 			return;
 		}
 
-		tribe_singleton( 'tickets.attendee_registration', 'Tribe__Tickets__Attendee_Registration__Main' );
-		tribe_singleton( 'tickets.attendee_registration.template', 'Tribe__Tickets__Attendee_Registration__Template' );
-		tribe_singleton( 'tickets.attendee_registration.view', 'Tribe__Tickets__Attendee_Registration__View' );
-		tribe_singleton( 'tickets.attendee_registration.rewrite', 'Tribe__Tickets__Attendee_Registration__Rewrite' );
-		tribe_singleton( 'tickets.attendee_registration.meta', 'Tribe__Tickets__Attendee_Registration__Meta' );
-		tribe_singleton( 'tickets.attendee_registration.shortcode', 'Tribe__Tickets__Attendee_Registration__Shortcode' );
-		tribe_singleton( 'tickets.attendee_registration.modal', 'Tribe__Tickets__Attendee_Registration__Modal' );
+		// After priority 45 from register() using this same action.
+		add_action( 'tribe_plugins_loaded', [ $this, 'add_attendee_registration_template_hook' ], 50 );
+
+		add_action( 'init', [ $this, 'add_attendee_registration_shortcode_hook' ] );
```

Comment:
Why aren't we using the callback way that we had before?

Reply:
If we want to use the container, we can use `$this->container->callback( '....', 'hook' )`

Reply:
I did this with @bordoni screen sharing. Late (just in time) instantiation.